prov/udp/src/udpx_attr.c

```diff
@@ -93,6 +94,39 @@ struct fi_info udpx_info = {
 
 struct util_prov udpx_util_prov = {
 	.prov = &udpx_prov,
-	.info = &udpx_info,
-	.flags = 0,
+	.info = NULL,
+        .flags = 0,
 };
+
+static int detect_mtu(const struct fi_info* info) {
+
+	struct ifaddrs ifaddrs;
+	ifaddrs.ifa_next = NULL;
+	ifaddrs.ifa_flags = 0;
+	ifaddrs.ifa_netmask = NULL;
+	ifaddrs.ifa_name = info->domain_attr->name;
+	ifaddrs.ifa_addr = info->src_addr;
+	return ofi_ifaddr_get_mtu(&ifaddrs);
```

Comment:
The window version of `ofi_ifaddr_get_mtu` returns `ifaddrs.mtu` which is undefined here.

Reply:
Please see the updated version.